script/scriptgen.py

```diff
@@ -659,11 +659,19 @@ def gen_read_files(self, f):
                 suffix = ""
                 if repodir.attrib.get("suffix"):
                     suffix = repodir.attrib["suffix"]
+                repopath = ""
+                if "/" in self.folder:
+                    repopath = self.ag.productpath + "/" + self.folder + "/*" + repodir.attrib["folder"] + "*" + suffix
+                else:
+                    repopath = (
+                        self.ag.productrepopath() + "/" + self.folder + "/*" + repodir.attrib["folder"] + "*" + suffix
+                    )
```

Comment:
```suggestion
    if "/" in self.folder:
        prd_path = self.ag.productpath
    else:
       prd_path = self.ag.productrepopath()
       glob_pattern = f"*{repodir.attrib['folder']}*{suffix}"
       repopath = (os.path.join(prd_path, self.folder, glob_pattern))
```
I think this reduce the duplication